File: backend/src/main/java/ropold/backend/model/LocationModel.java
package ropold.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "locations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LocationModel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "address", columnDefinition = "TEXT")
    private String address;

    @Column(name = "phone")
    private String phone;

    @Column(name = "email")
    private String email;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;
}

File: frontend/src/components/models/LocationModel.ts

export type LocationModel = {
    id: string;
    name: string;
    address: string | null;
    phone: string | null;
    email: string | null;
    notes: string | null;
    latitude: number | null;
    longitude: number | null;
    imageUrl: string | null;
};

export const DefaultLocation: LocationModel = {
    id: "",
    name: "",
    address: null,
    phone: null,
    email: null,
    notes: null,
    latitude: null,
    longitude: null,
    imageUrl: null,
};